<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Controller</class>
 <widget class="QWidget" name="Controller">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Controller</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(24, 4, 31);</string>
  </property>
  <widget class="QLineEdit" name="le_neurons_first_layer">
   <property name="geometry">
    <rect>
     <x>349</x>
     <y>490</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>256</string>
   </property>
  </widget>
  <widget class="QLabel" name="l_answer">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>10</y>
     <width>71</width>
     <height>71</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(170, 1, 51);
color: white;
border-radius:20;</string>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:52pt;&quot;&gt;?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QGraphicsView" name="graphic_scene">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>420</width>
     <height>420</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
  </widget>
  <widget class="QPushButton" name="b_do_learn">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>259</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(25, 141, 99);
color: white;
border-radius:10;</string>
   </property>
   <property name="text">
    <string>go</string>
   </property>
  </widget>
  <widget class="QLabel" name="l_path_import">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>423</y>
     <width>341</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="b_create_AI">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>530</y>
     <width>261</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(51, 4, 45);
color: white;
border-radius:15;</string>
   </property>
   <property name="text">
    <string>create AI</string>
   </property>
  </widget>
  <widget class="QPushButton" name="b_do_export">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>532</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(25, 141, 99);
color: white;
border-radius:10;</string>
   </property>
   <property name="text">
    <string>go</string>
   </property>
  </widget>
  <widget class="QPushButton" name="b_path_import">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>448</y>
     <width>271</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(51, 4, 45);
color: white;
border-radius:15;</string>
   </property>
   <property name="text">
    <string>import</string>
   </property>
  </widget>
  <widget class="QLabel" name="l_path_learn">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>184</y>
     <width>341</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="b_do_import">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>448</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(25, 141, 99);
color: white;
border-radius:10;</string>
   </property>
   <property name="text">
    <string>go</string>
   </property>
  </widget>
  <widget class="QLabel" name="l_path_export">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>507</y>
     <width>341</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="b_path_export">
   <property name="geometry">
    <rect>
     <x>449</x>
     <y>532</y>
     <width>271</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(51, 4, 45);
color: white;
border-radius:15;</string>
   </property>
   <property name="text">
    <string>export</string>
   </property>
  </widget>
  <widget class="QPushButton" name="b_predict">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>100</y>
     <width>271</width>
     <height>70</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(51, 4, 45);
color: white;
border-radius:15;</string>
   </property>
   <property name="text">
    <string>predict</string>
   </property>
  </widget>
  <widget class="QPushButton" name="b_do_test">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>346</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(25, 141, 99);
color: white;
border-radius:10;</string>
   </property>
   <property name="text">
    <string>go</string>
   </property>
  </widget>
  <widget class="QLabel" name="l_path_test">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>321</y>
     <width>341</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QGroupBox" name="gr_data_mode">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>464</y>
     <width>120</width>
     <height>80</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="title">
    <string>mode</string>
   </property>
   <widget class="QRadioButton" name="rb_matrix_mode">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>35</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Papyrus</family>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>matrix</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="rb_graph_mode">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>55</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Papyrus</family>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>graph</string>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="b_path_learn">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>209</y>
     <width>171</width>
     <height>91</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(51, 4, 45);
color: white;
border-radius:15;</string>
   </property>
   <property name="text">
    <string>learn</string>
   </property>
  </widget>
  <widget class="QPushButton" name="b_path_test">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>346</y>
     <width>271</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(51, 4, 45);
color: white;
border-radius:15;</string>
   </property>
   <property name="text">
    <string>test</string>
   </property>
  </widget>
  <widget class="QComboBox" name="cb_count_hidden_layers">
   <property name="geometry">
    <rect>
     <x>345</x>
     <y>445</y>
     <width>81</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <item>
    <property name="text">
     <string>2</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>3</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>4</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>5</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="l_count_layers">
   <property name="geometry">
    <rect>
     <x>163</x>
     <y>450</y>
     <width>171</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;hidden layers&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="l_neuron_first_layer">
   <property name="geometry">
    <rect>
     <x>163</x>
     <y>490</y>
     <width>171</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;neurons in 1 hid layer&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QSlider" name="s_test_proc">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>388</y>
     <width>301</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">color: rgb(11, 142, 98);</string>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
   <property name="singleStep">
    <number>1</number>
   </property>
   <property name="pageStep">
    <number>1</number>
   </property>
   <property name="value">
    <number>100</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="invertedAppearance">
    <bool>false</bool>
   </property>
   <property name="invertedControls">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLabel" name="l_test_slider">
   <property name="geometry">
    <rect>
     <x>750</x>
     <y>390</y>
     <width>31</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;1&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="le_learn_epoch">
   <property name="geometry">
    <rect>
     <x>706</x>
     <y>264</y>
     <width>21</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>1</string>
   </property>
  </widget>
  <widget class="QPushButton" name="b_open_img">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>100</y>
     <width>61</width>
     <height>70</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(25, 141, 99);
color: white;
border-radius:10;</string>
   </property>
   <property name="text">
    <string>open
.bmp</string>
   </property>
  </widget>
  <widget class="QPushButton" name="b_do_cross">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>210</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(25, 141, 99);
color: white;
border-radius:10;</string>
   </property>
   <property name="text">
    <string>go</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="le_cross_valid">
   <property name="geometry">
    <rect>
     <x>706</x>
     <y>215</y>
     <width>21</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>3</string>
   </property>
  </widget>
  <widget class="QLabel" name="l_learn_epoch">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>259</y>
     <width>68</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;epoches&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="l_cross_valid">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>210</y>
     <width>71</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string>     cross
 validation</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>20</y>
     <width>141</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Papyrus</family>
     <pointsize>23</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(48, 48, 48);
border-radius:8;</string>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;AI's answer:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
